tensorflow/python/ops/array_ops.py

```diff
@@ -4352,6 +4352,18 @@ def where_v2(condition, x=None, y=None, name=None):
   <tf.Tensor: shape=(4,), dtype=int32, numpy=array([100, 100, 100, 100],
   dtype=int32)>
 
+  Note that if any of the input to a tf.where contains 'NaN's, then the gradient will always
```

Comment:
That's not true.

The issue is more like 

Please note that if the gradient of either branch of the `tf.where` generates a NaN then the gradient of the entire `tf.where` will be NaN. For example, `tf.where(y > 0, tf.sqrt(y), y)` has a NaN gradient if `y` has any negative numbers in it. To avoid the NaN you need to avoid computing any operation which might have a NaN gradient, so instead do `tf.where(y > 0, tf.sqrt(tf.where(y > 0, y, 1)), y)` (essentially using `tf.where` again to avoid computing a value whose gradient is NaN by replacing dangerous inputs with safe inputs).